sql/planbuilder/set.go

```diff
@@ -243,16 +244,19 @@ func (b *Builder) simplifySetExpr(name *ast.ColName, varScope ast.SetScope, val
 			return nil, false
 		}
 
-		switch strings.ToLower(setVal) {
-		case ast.KeywordString(ast.ON):
-			return expression.NewLiteral(true, types.Boolean), true
-		case ast.KeywordString(ast.TRUE):
-			return expression.NewLiteral(true, types.Boolean), true
-		case ast.KeywordString(ast.OFF):
-			return expression.NewLiteral(false, types.Boolean), true
-		case ast.KeywordString(ast.FALSE):
-			return expression.NewLiteral(false, types.Boolean), true
-		default:
+		// If we're targeting a boolean type (likely if we see an INT8 sys var),
+		// convert a few common string values to boolean values.
+		if sysVarType != nil && sysVarType.Type() == sqltypes.Int8 {
```

Comment:
Can move `nil` check from below up to save on one if condition.

Reply:
Good eye! Thanks for spotting that.